backend_plugin/src/main/evaluators/AddImportEvaluator.java

```diff
@@ -10,102 +10,102 @@
+		} catch (BadLocationException e) {
+		}
+		return false;
+	}
+
+	/**
+	 * Checks whether a document change event is occurring on a line that did
+	 * not have an import statement previously
+	 * @param event
+	 * @return false
+	 */
+	public boolean evaluateDocumentChanges(DocumentEvent event) {
+		try {
+
+			// If the line before the change was not an import statement, but the
+			// line after the change is, and there are unresolved variables, return true
+			int line = document.getLineOfOffset(event.getOffset());
+			return this.unresolvedVariablesExist &&
+					!this.lineHadImportStatementAlready && lineIsAnImportStatement(line);
+		} catch (BadLocationException e) {
+		}
+		return false;
 	}
-	return false;
-    }
-
-    /**
-     * Updates the boolean flag of this AddImportEvaluator that tracks whether
-     * there exist any unresolved types in the document. This returns false
-     * always, as a change to the annotation model should not by itself trigger
-     * a notification to the frontend.
-     * @return boolean false
-     */
-    @Override
-    public boolean evaluateAnnotationModelChanges(IAnnotationModel model) {
-	Iterator<Annotation> it = model.getAnnotationIterator();
-
-	// Iterate through all annotations
-	while (it.hasNext()) {
-	    Annotation current = it.next();
-
-	    // If the annotation is valid and represents an unresolved variable,
-	    // update the flag for unresolved variables existing
-	    if (current.getText().endsWith("cannot be resolved to a type") && !current.isMarkedDeleted()) {
-		this.unresolvedVariablesExist = true;
+
+	/**
+	 * Checks if the given line begins with an import statement
+	 * @param line
+	 * @return
+	 */
+	public boolean lineIsAnImportStatement(int line) {
+		try {
+
+			int startOffset = document.getLineOffset(line);
+			int length = document.getLineLength(line);
+
+			// Use regex to remove only the leading white space from the line,
+			// then check if the line starts with "import "
+			String lineContents = document.get(startOffset, length);
+			return lineContents.replaceAll("^\\s+", "").startsWith("import ");
+		} catch (BadLocationException e) {
+		}
+		return false;
+	}
+
+	/**
+	 * Updates the boolean flag of this AddImportEvaluator that tracks whether
+	 * there exist any unresolved types in the document. This returns false
+	 * always, as a change to the annotation model should not by itself trigger
+	 * a notification to the frontend.
+	 * @return boolean false
+	 */
+	@Override
+	public boolean evaluateAnnotationModelChanges(IAnnotationModel model) {
+		Iterator<Annotation> it = model.getAnnotationIterator();
+
+		// Iterate through all annotations
+		while (it.hasNext()) {
+			Annotation current = it.next();
+
+			// If the annotation is valid and represents an unresolved variable,
+			// update the flag for unresolved variables existing
+			if (current.getText().endsWith("cannot be resolved to a type") && !current.isMarkedDeleted()) {
```

Comment:
One more thing here. current.getText() can be null and throw a null pointer exception. It's what caused the exceptions when I was trying to demo our current build. I fixed the current build by adding 
```suggestion
			if (current.getText != null && current.getText().endsWith("cannot be resolved to a type") && !current.isMarkedDeleted()) {
```

Reply:
Nice catch, I've updated the PR again